docs/topics/OAuth2.md

```diff
@@ -56,68 +41,229 @@ Scopes provide access to certain resources of a user's account. Your API client
+2. Bots cannot have friends, nor be added to or join Group DMs.
+3. Bots do not have a maximum number of Guilds (unlike user accounts, which are limited to 100).
+4. Bots have an entirely separate set of [Rate Limits](#DOCS_RATE_LIMITS/rate-limits).
+
+### Bot Auth Flow
+
+Bot auth is a special server-less and callback-less OAuth2 flow that makes it easy for users to add bots to servers. The URL you create looks similar to what we use for full stack implementation:
+
+```
+https://discordapp.com/oauth2/authorize?client_id=157730590492196864&scope=bot&permissions=1
 ```
 
-`client_id` is your application's ID and `redirect_uri` is one of your application's URL-encoded redirect URIs.
+In the case of bots, the `scope` parameter should be set to `bot`. There's also a new parameter, `permissions`, which is an integer corresponding to the [permission calculations](#DOCS_PERMISSIONS/permissions-bitwise-permission-flags) for the bot. You'll also notice the absence of `response_type` and `redirect_uri`. Bot auth does not require these parameters because there is no need to retrieve the user's access token.
+
+When the user navigates to this page, they'll be prompted to add the bot to a server in which they have proper permissions. On acceptance, the bot will be added. Super easy!
+
+### Advanced Bot Auth
 
-When a user is directed to this URL, they are prompted to select a channel for the webhook to be placed in. Your application will receive an authorization code back in the querystring (as usual with the authorization code grant).
+Enterprising devs can add some complexity to bot auth. You can request additional scopes outside of "bot", which will prompt a continuation into a complete flow and add the ability to request the user's access token. If you request any scopes outside of `bot`, `response_type` is again mandatory; we will also automatically redirect the user to the first uri in your application's registered list unless `redirect_uri` is specified. A full OAuth2 flow is also required if you check "Require OAuth2 Code Grant" in your application's settings.
 
-When you exchange the authorization code for an access token, the token response will contain the [webhook](#DOCS_WEBHOOK/webhook-object) object:
+If you continue into the full stack flow while including "bot" in your scopes, you'll get some additional querystring parameters on redirection: `guild_id`, the id of the guild to which the bot was added, and `permissions`, the permission integer from the original URL. `guild_id` should only be used as a hint that the bot has joined that guild and not as a source of truth. If you are looking to associate your bot to that guild, you should ensure at minimum that the user adding the bot has MANAGE_SERVER permissions on that guild. You could also use the [Get Current User Guilds](#DOCS_USER/get-current-user-guilds) endpoint with your bot's token and look for that `guild_id`.
```

Comment:
`bot` over "bot"

Reply:
> If you are looking to associate your bot to that guild, you should ensure at minimum that the user adding the bot has MANAGE_SERVER permissions on that guild.

I think you misunderstood the purpose of the warning. If you need to associate a user account on your service (not on Discord) with a Discord guild, you can only do that by ensuring the user had permission to add the bot (adding a bot requires MANAGE_GUILD permission).

Reply:
> You could also use the [Get Current User Guilds](#DOCS_USER/get-current-user-guilds) endpoint

that doesn't really fix the issue and that endpoint probably fails to load for large bots. basically the only current way to validate the guild id passed as qs would be to get the user's id with oauth2 bearer token and then grab their permissions on the guild to check for `MANAGE_GUILD`.. so that means you grab the member and also the guild and compute permissions for that member.

this seems really terrible and would be easily solved by simply returning guild information in the token request response.. that's an easy change, will do it this coming week.. then we can document to use the `full OAuth2 authorization code grant flow` for this solving this case .. i really dislike that we ever thought it was a good idea to return that information in the querystring.. we should honestly deprecate and remove it